tests/chainer_tests/links_tests/loss_tests/test_negative_sampling.py

```diff
@@ -34,14 +30,49 @@ def setUp(self):
             g_shape = ()
         self.gy = numpy.random.uniform(-1, 1, g_shape).astype(numpy.float32)
 
-    def check_forward(self, x_data, t_data):
+    def create_link(self, rng=None):
+        if rng is None:
+            rng = numpy.random.RandomState()
+        link = links.NegativeSampling(
+            self.in_size, [10, 5, 2, 5, 2], self.sample_size)
+        link.cleargrads()
+        # W is initialized with zero. Inject random values for meaningful test.
+        link.W.array[:] = rng.uniform(-1, 1, link.W.shape)
+        return link
+
+    def call_link_and_return_samples(self, func):
+        # Calls the link while sneaking the samples returned from
+        # F.negative_sampling.
+        # func is a function in which the link is called.
+
+        # Wrap F.negative_sampling to sneak the samples.
+        orig_negative_sampling = chainer.functions.negative_sampling
+        saved_samples = [None]
+
+        def wrap_negative_sampling(*args, **kwargs):
+            out, samples = orig_negative_sampling(
+                *args, return_samples=True, **kwargs)
+            saved_samples[0] = samples
+            return out
+
+        with testing.patch(
+                'chainer.functions.loss.negative_sampling.negative_sampling',
+                wraps=wrap_negative_sampling) as m:
+            out = func()
+            assert m.call_count == 1
+
+        assert saved_samples[0] is not None
+        return out, saved_samples[0]
+
+    def check_forward(self, link, x_data, t_data):
         x = chainer.Variable(x_data)
         t = chainer.Variable(t_data)
-        y = self.link(x, t, reduce=self.reduce)
+        y, samples = self.call_link_and_return_samples(
+            lambda: link(x, t, reduce=self.reduce))
```

Comment:
How about adding `return_samples` argument to `L.NegativeSampling.forward`? I like such a consistency and we can avoid the patching trick in the tests.

Reply:
Added the option as suggested.